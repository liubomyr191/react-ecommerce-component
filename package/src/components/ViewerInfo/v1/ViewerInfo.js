import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { withComponents } from "@reactioncommerce/components-context";
import { addTypographyStyles, applyTheme } from "../../../utils";

const ViewerInfoContainer = styled.div`
  display: flex;
  position: relative;
`;

const ViewerFirstNameText = styled.span`
  ${addTypographyStyles("ViewerInfo", "labelText")}
  display: ${({ compact, full }) => {
    if (full) {
      return compact ? "none" : "inline";
    }
    return "none";
  }};
  align-self: center;
  margin-left: 0.5rem;

  @media (min-width: ${applyTheme("md", "breakpoints")}px) {
    display: ${({ compact }) => (compact ? "none" : "inline")};
  }
`;

class ViewerInfo extends Component {
  static propTypes = {
    /**
     * Enable this prop when you only want to display the initials/avatar on all screens
     */
    compact: PropTypes.bool,
    /**
     * If you've set up a components context using
     * [@reactioncommerce/components-context](https://github.com/reactioncommerce/components-context)
     * (recommended), then this prop will come from there automatically. If you have not
     * set up a components context or you want to override one of the components in a
     * single spot, you can pass in the components prop directly.
     */
    components: PropTypes.shape({
      /**
     * Profile image component to display
     */
      ProfileImage: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
    }),
    /**
     * Enable this prop when you want to display the initials and first name on all screens
     */
    full: PropTypes.bool,
    /**
    * An object containing basic user information.
    */
    viewer: PropTypes.shape({
      firstName: PropTypes.string,
      lastName: PropTypes.string,
      name: PropTypes.string,
      primaryEmailAddress: PropTypes.string.isRequired,
      profileImage: PropTypes.string
    }).isRequired
  };

  static defaultProps = {
    compact: false,
    full: false
  };

  /**
   *
   * @name viewerName
   * @summary If `firstName` is available on the `viewer` object, return that.
   *   Otherwise return the email address.
   * @return {String} Display name for the viewer
   */
  get viewerName() {
    const { viewer: { firstName, primaryEmailAddress } } = this.props;
    return (firstName && firstName) || primaryEmailAddress;
  }

  render() {
    const { compact, components: { ProfileImage }, full, viewer } = this.props;
    return (
      <ViewerInfoContainer>
        <ProfileImage size={30} viewer={viewer} />
        <ViewerFirstNameText compact={compact} full={full}>
          {this.viewerName}
        </ViewerFirstNameText>
      </ViewerInfoContainer>
    );
  }
}

export default withComponents(ViewerInfo);
