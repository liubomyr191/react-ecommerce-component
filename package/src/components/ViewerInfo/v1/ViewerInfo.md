### Overview
Renders a user's first name next to their initials.

#### Usage
```jsx
<ViewerInfo />
```

#### With Viewer
```jsx
const viewer = {
  firstName: "Issac",
  lastName: "Newton"
};

<ViewerInfo viewer={viewer} />
```

**Long Name**
```jsx
const viewer = {
  firstName: "Madhavaditya",
  lastName: "Balakrishnan"
};

<ViewerInfo viewer={viewer} />
```

**Hypenated Name**
```jsx
const viewer = {
  firstName: "Keeanga-Yamahtta",
  lastName: "Taylor"
};

<ViewerInfo viewer={viewer} />
```

**Incomplete Name**
```jsx
const viewer = {
  firstName: "Olamide"
};

<ViewerInfo viewer={viewer} />
```

#### Full
```jsx
const viewer = {
  firstName: "Ligaya",
  lastName: "Ocampo"
};

<ViewerInfo viewer={viewer} full />
```

#### Compact
```jsx
const viewer = {
  firstName: "Patricia",
  lastName: "Smith"
};

<ViewerInfo viewer={viewer} compact />
```
