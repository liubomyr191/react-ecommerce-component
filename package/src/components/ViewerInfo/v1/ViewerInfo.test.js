import React from "react";
import renderer from "react-test-renderer";
import ViewerInfo from "./ViewerInfo";

test("Render a viewerinfo without any props", () => {
  const component = renderer.create(<ViewerInfo />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Render a viewer's initials and first name", () => {
  const viewer = {
    firstName: "Issac",
    lastName: "Newton"
  };

  const component = renderer.create(<ViewerInfo viewer={viewer} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
