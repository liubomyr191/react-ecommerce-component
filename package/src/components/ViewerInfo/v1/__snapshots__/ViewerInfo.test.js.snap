// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a viewer's initials and first name 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  background: #8ce0c9;
  width: 30px;
  height: 30px;
  border-radius: 50%;
  display: inline-block;
}

.c2 {
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-weight: 400;
  font-size: 14px;
  color: #ffffff;
  position: absolute;
  top: 6px;
  left: 9px;
  z-index: 10;
}

.c3 {
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 14px;
  color: #3c3c3c;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-left: 0.5rem;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  >
    IN
  </div>
  <span
    className="c3"
  >
    Issac
  </span>
</div>
`;
